app/src/main-process/menu/build-default-menu.ts

```diff
@@ -21,6 +21,12 @@ const defaultBranchNameDefaultValue = __DARWIN__
   ? 'Default Branch'
   : 'default branch'
 
+enum ZoomDirection {
```

Comment:
This move was required because we have a `typescript/no-use-before-define` rule setup, but up until now the parser didn't identify this as a problem.

Reply:
Personally, I’m not a big fan of `no-use-before-define` since I feel it’s better to order things in a way that makes sense rather than by dependency.

Reply:
This is an interesting case because `build-default-menu` is ~540 lines of JS and the zoom function was just added to the bottom of the file out of convenience, but rather than abandon the rule we could extract the zoom-specific logic out to it's own module to keep the things `zoom()` needs in one place:

```diff
diff --git a/app/src/main-process/menu/build-default-menu.ts b/app/src/main-process/menu/build-default-menu.ts
index a38db88a4..6737676d2 100644
--- a/app/src/main-process/menu/build-default-menu.ts
+++ b/app/src/main-process/menu/build-default-menu.ts
@@ -7,6 +7,8 @@ import { ensureDir } from 'fs-extra'
 
 import { log } from '../log'
 import { openDirectorySafe } from '../shell'
+import { ClickHandler } from '../models/click-handler'
+import { zoom, ZoomDirection } from './zoom'
 
 const defaultEditorLabel = __DARWIN__
   ? 'Open in External Editor'
@@ -453,12 +455,6 @@ export function buildDefaultMenu({
   return Menu.buildFromTemplate(template)
 }
 
-type ClickHandler = (
-  menuItem: Electron.MenuItem,
-  browserWindow: Electron.BrowserWindow,
-  event: Electron.Event
-) => void
-
 /**
  * Utility function returning a Click event handler which, when invoked, emits
  * the provided menu event over IPC.
@@ -472,68 +468,3 @@ function emit(name: MenuEvent): ClickHandler {
     }
   }
 }
-
-enum ZoomDirection {
-  Reset,
-  In,
-  Out,
-}
-
-/** The zoom steps that we support, these factors must sorted */
-const ZoomInFactors = [1, 1.1, 1.25, 1.5, 1.75, 2]
-const ZoomOutFactors = ZoomInFactors.slice().reverse()
-
-/**
- * Returns the element in the array that's closest to the value parameter. Note
- * that this function will throw if passed an empty array.
- */
-function findClosestValue(arr: Array<number>, value: number) {
-  return arr.reduce((previous, current) => {
-    return Math.abs(current - value) < Math.abs(previous - value)
-      ? current
-      : previous
-  })
-}
-
-/**
- * Figure out the next zoom level for the given direction and alert the renderer
- * about a change in zoom factor if necessary.
- */
-function zoom(direction: ZoomDirection): ClickHandler {
-  return (menuItem, window) => {
-    if (!window) {
-      return
-    }
-
-    const { webContents } = window
-
-    if (direction === ZoomDirection.Reset) {
-      webContents.setZoomFactor(1)
-      webContents.send('zoom-factor-changed', 1)
-    } else {
-      webContents.getZoomFactor(rawZoom => {
-        const zoomFactors =
-          direction === ZoomDirection.In ? ZoomInFactors : ZoomOutFactors
-
-        // So the values that we get from getZoomFactor are floating point
-        // precision numbers from chromium that don't always round nicely so
-        // we'll have to do a little trick to figure out which of our supported
-        // zoom factors the value is referring to.
-        const currentZoom = findClosestValue(zoomFactors, rawZoom)
-
-        const nextZoomLevel = zoomFactors.find(f =>
-          direction === ZoomDirection.In ? f > currentZoom : f < currentZoom
-        )
-
-        // If we couldn't find a zoom level (likely due to manual manipulation
-        // of the zoom factor in devtools) we'll just snap to the closest valid
-        // factor we've got.
-        const newZoom =
-          nextZoomLevel === undefined ? currentZoom : nextZoomLevel
-
-        webContents.setZoomFactor(newZoom)
-        webContents.send('zoom-factor-changed', newZoom)
-      })
-    }
-  }
-}
diff --git a/app/src/main-process/menu/zoom.ts b/app/src/main-process/menu/zoom.ts
new file mode 100644
index 000000000..fcef3c6c1
--- /dev/null
+++ b/app/src/main-process/menu/zoom.ts
@@ -0,0 +1,65 @@
+import { ClickHandler } from '../models/click-handler'
+
+export enum ZoomDirection {
+  Reset,
+  In,
+  Out,
+}
+
+/** The zoom steps that we support, these factors must sorted */
+const ZoomInFactors = [1, 1.1, 1.25, 1.5, 1.75, 2]
+const ZoomOutFactors = ZoomInFactors.slice().reverse()
+
+/**
+ * Returns the element in the array that's closest to the value parameter. Note
+ * that this function will throw if passed an empty array.
+ */
+function findClosestValue(arr: Array<number>, value: number) {
+  return arr.reduce((previous, current) => {
+    return Math.abs(current - value) < Math.abs(previous - value)
+      ? current
+      : previous
+  })
+}
+/**
+ * Figure out the next zoom level for the given direction and alert the renderer
+ * about a change in zoom factor if necessary.
+ */
+export function zoom(direction: ZoomDirection): ClickHandler {
+  return (menuItem, window) => {
+    if (!window) {
+      return
+    }
+
+    const { webContents } = window
+
+    if (direction === ZoomDirection.Reset) {
+      webContents.setZoomFactor(1)
+      webContents.send('zoom-factor-changed', 1)
+    } else {
+      webContents.getZoomFactor(rawZoom => {
+        const zoomFactors =
+          direction === ZoomDirection.In ? ZoomInFactors : ZoomOutFactors
+
+        // So the values that we get from getZoomFactor are floating point
+        // precision numbers from chromium that don't always round nicely so
+        // we'll have to do a little trick to figure out which of our supported
+        // zoom factors the value is referring to.
+        const currentZoom = findClosestValue(zoomFactors, rawZoom)
+
+        const nextZoomLevel = zoomFactors.find(f =>
+          direction === ZoomDirection.In ? f > currentZoom : f < currentZoom
+        )
+
+        // If we couldn't find a zoom level (likely due to manual manipulation
+        // of the zoom factor in devtools) we'll just snap to the closest valid
+        // factor we've got.
+        const newZoom =
+          nextZoomLevel === undefined ? currentZoom : nextZoomLevel
+
+        webContents.setZoomFactor(newZoom)
+        webContents.send('zoom-factor-changed', newZoom)
+      })
+    }
+  }
+}
diff --git a/app/src/main-process/models/click-handler.ts b/app/src/main-process/models/click-handler.ts
new file mode 100644
index 000000000..6216daf47
--- /dev/null
+++ b/app/src/main-process/models/click-handler.ts
@@ -0,0 +1,5 @@
+export type ClickHandler = (
+  menuItem: Electron.MenuItem,
+  browserWindow: Electron.BrowserWindow,
+  event: Electron.Event
+) => void
```